### DISCUSSION TITLE
Deploy Payment Connector with commerce SDK

### DISCUSSION THREAD

**Discussion by ahgamal-ms** [microsoft/Dynamics365Commerce.Solutions#176] [Q&A] [answered]
Customer has picked the payment connector sample from retail SDK and implemented with commerce SDK , now we have added inside the scale unit deployable project a reference to payment connector project.
   
We have also tried to include the payment connector dlls with content include however the CSU package Is created with out payment connector dll.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/74003164/257e9713-e156-4d48-8479-8cadc026fb6a)

So do we have an guidance on how customer can do that with commerce SKD  , I have found this link but seems more retailed to retail SDK https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/deploy-payment-connector

**Comment by madyke** [comment]
Hello @ahgamal-ms - the process should be the same as including files like in other Commerce SDK extensions. I can see that the `GG.ScaleUnit.PaymentConnectors` project has a reference to the Klarna payment connector project, and it looks like the payment connector DLL's are being Content Included correctly.

Just to be sure - does the `GG.ScaleUnit.Installer` project have a reference to this (the `GG.ScaleUnit.PaymentConnectors`) project?

**Reply by ahgamal-ms** [reply]
Hello @madyke. the customer uses the scale unit project to create the package to CSU in LCS and not the. installer project.
so for the scale unit project the connector is referenced as below.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/74003164/3c3fe608-d892-420e-9e6c-e140b7552b65)

And if we open the output of GG.ScaleUnit, and browse into the ext folder, we see the CPOKlarnaPaymentConnector was not added.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/74003164/bd559bef-1ef4-4979-a2aa-a229b10907a0)

**Comment by madyke** [comment]
Hi @ahgamal-ms - I'll check with the CSU team to see if there are different requirements for the payment connectors in their package, though don't think there would be.

If the customer is using `GG.ScaleUnit` to create the package and they expect the `CPOKlarnaPaymentConnector.dll` to be in that project's bin folder, are they Content Including the .dll in the `GG.ScaleUnit.csproj` as well? The screenshot from the initial post does show the .dll being Content Included, but it looks like thats in the `GG.ScaleUnit.PaymentConnector` project, not the `GG.ScaleUnit` project.

**Reply by ahgamal-ms** [reply]
@madyke  still did not have any impact, sadly.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/74003164/0c79fd8b-75f0-4596-8f89-19fb8f416ec2)

**Reply by madyke** [reply, marked as answer]
Hi @ahgamal-ms - I discussed with the team and it looks like CSU does actually have some different rules around their targets file. We've filed a documentation bug so that customers can see exactly which ItemGroups they should be using for each feature area.

For now though, in the `GG.ScaleUnit` project can you try replacing the `Content` tag on lines 20 and 22 with `IPayment_Processor_Assemblies` instead? It looks like that should resolve the issue and copy the files into the correct folder location. So the full code block would be
```
<ItemGroup>
  <IPayment_Processor_Assemblies Include="path to dll">
    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
  </IPayment_Processor_Assemblies>
</ItemGroup>
```

**Reply by ahgamal-ms** [reply]
Thanks @madyke  The payment connector now appears in the ext-folder.

However this makes me wonder/worry a little, because it is my understanding that this files should be put in the Connectors-directory ? Can you verify the file will be loaded if it is instead located in the ext-folder ?

**Reply by madyke** [reply]
Hi @ahgamal-ms - The `Connectors` directory is only for the out-of-the-box connectors that Microsoft provides, any files/folders that the customer adds there may not even be picked up during build time.

Assemblies for payment connectors and extensions that the customer is adding should be included in the `Ext` folder, CSU will look for them there.
